src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingService.java

```diff
@@ -304,15 +308,22 @@ private void updateNotification() {
             if (type.equals("low_priority"))
                 builder.setPriority(NotificationCompat.PRIORITY_MIN);
 
-            nm.notify(NOTIFICATION_ACTIVE, builder.build());
+            if (foreground) {
+                builder.setContentText(getString(R.string.syncthing_active_foreground));
+                startForeground(NOTIFICATION_ACTIVE, builder.build());
+            } else {
+                stopForeground(false); // ensure no longer running with foreground priority
+                nm.notify(NOTIFICATION_ACTIVE, builder.build());
+            }
         } else {
+            stopForeground(false); // ensure no longer running with foreground priority
```

Comment:
I put comments on their own line before the code, I think that's easier to see. But I don't have a strong strong preference either way ...